src/Collectors/JobCollector.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace AG\ElasticApmLaravel\Collectors;
+
+use AG\ElasticApmLaravel\Agent;
+use AG\ElasticApmLaravel\Collectors\Interfaces\DataCollectorInterface;
+use Illuminate\Foundation\Application;
+use Illuminate\Queue\Events\JobProcessed;
+use Illuminate\Queue\Events\JobProcessing;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Log;
+use PhilKra\Events\Transaction;
+use Throwable;
+
+/**
+ * Collects info about the job process.
+ */
+class JobCollector extends TimelineDataCollector implements DataCollectorInterface
+{
+    protected $app;
+    protected $agent;
+
+    public function __construct(Application $app, Agent $agent, float $request_start_time)
+    {
+        parent::__construct($request_start_time);
+
+        $this->app = $app;
+        $this->agent = $agent;
+        $this->registerEventListeners();
+    }
+
+    public static function getName(): string
+    {
+        return 'job-collector';
+    }
+
+    protected function registerEventListeners(): void
+    {
+        $this->app->events->listen(JobProcessing::class, function (JobProcessing $event) {
+            $this->startTransaction($this->getTransactionName($event));
+        });
+
+        $this->app->events->listen(JobProcessed::class, function (JobProcessed $event) {
+            $transaction_name = $this->getTransactionName($event);
+            $this->addMetadata($transaction_name, $event->job);
+            $this->stopTransaction($transaction_name);
+        });
+    }
+
+    protected function startTransaction(string $transaction_name): Transaction
+    {
+        return $this->agent->startTransaction(
+            $transaction_name,
+            [],
+            $this->request_start_time
+        );
+    }
+
+    protected function addMetadata($transaction_name, $job): void
```

Comment:
I left `$job` intentionally but will document why--this doesn't do much by default so it provides a bit more info for someone who might extend this method. Let me know if you'd still like it removed.

Reply:
I'm afraid the linter will complain. Maybe we could actually use the variable to add metadata, like the number of retries of the queue name.

Reply:
It hasn't yet, I don't think php-cs-fixer has that rule as an option. PHPMD does though. https://phpmd.org/rules/unusedcode.html

Reply:
I went ahead and made a change to log the result just like HTTP does. I didn't know what else to do so I went with 200 for success and 400 for failure since it's familiar.